File: python-sdk/src/boxsdk/models/shared/accesstoken.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils
from ..shared import filescope as shared_filescope

class AccessTokenIssuedTokenTypeEnum(str, Enum):
    URN_IETF_PARAMS_OAUTH_TOKEN_TYPE_ACCESS_TOKEN = "urn:ietf:params:oauth:token-type:access_token"

class AccessTokenTokenTypeEnum(str, Enum):
    BEARER = "bearer"


@dataclass_json
@dataclasses.dataclass
class AccessToken:
    r"""AccessToken
    A token that can be used to make authenticated API calls.
    """
    
    access_token: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('access_token') }})
    expires_in: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('expires_in') }})
    issued_token_type: Optional[AccessTokenIssuedTokenTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('issued_token_type') }})
    refresh_token: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('refresh_token') }})
    restricted_to: Optional[list[shared_filescope.FileScope]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('restricted_to') }})
    token_type: Optional[AccessTokenTokenTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('token_type') }})
    

File: java-sdk/lib/src/main/java/models/shared/AccessToken.java
package .models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * AccessToken
 * A token that can be used to make authenticated API calls.
**/
public class AccessToken {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("access_token")
    public String accessToken;
    public AccessToken withAccessToken(String accessToken) {
        this.accessToken = accessToken;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("expires_in")
    public Long expiresIn;
    public AccessToken withExpiresIn(Long expiresIn) {
        this.expiresIn = expiresIn;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("issued_token_type")
    public AccessTokenIssuedTokenTypeEnum issuedTokenType;
    public AccessToken withIssuedTokenType(AccessTokenIssuedTokenTypeEnum issuedTokenType) {
        this.issuedTokenType = issuedTokenType;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("refresh_token")
    public String refreshToken;
    public AccessToken withRefreshToken(String refreshToken) {
        this.refreshToken = refreshToken;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("restricted_to")
    public FileScope[] restrictedTo;
    public AccessToken withRestrictedTo(FileScope[] restrictedTo) {
        this.restrictedTo = restrictedTo;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("token_type")
    public AccessTokenTokenTypeEnum tokenType;
    public AccessToken withTokenType(AccessTokenTokenTypeEnum tokenType) {
        this.tokenType = tokenType;
        return this;
    }
}